src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackEncoder.cs

```diff
@@ -463,16 +467,28 @@ public static bool EncodeStringLiteral(string value, Span<byte> destination, out
             // |  String Data (Length octets)  |
             // +-------------------------------+
 
-            if (destination.Length != 0)
+            if (destination.Length > value.Length)
```

Comment:
It is possible for encoded length to be less than `value.Length`. `!= 0` seems appropriate here.

Reply:
Changed back to `!= 0`